### PR TITLE
Support for Linux platform

### PR DESCRIPTION
**Pull request by FaerHack**
I borrowed cmake infra from [stevenhoving:cmake](https://github.com/stevenhoving/raw_pdb/tree/cmake)
Not sure if it can be merged but hope it will be useful for someone

### CODE REVIEW

**Comment by stevenhoving** [comment]
Does this fixes https://github.com/MolecularMatters/raw_pdb/issues/9 ?

File: src/Examples/ExampleContributions.cpp [hunk 1]
```diff
@@ -85,7 +85,7 @@ void ExampleContributions(const PDB::RawFile& rawPdbFile, const PDB::DBIStream&
 	{
 		printf("20 largest contributions:\n");
 
-		const size_t countToShow = std::min(20ull, contributions.size());
+		const size_t countToShow = std::min(static_cast<size_t>(20ul), contributions.size());
```
Comment (MolecularMatters) [inline src/Examples/ExampleContributions.cpp:88]: I prefer to use std::min<size_t>(arg1, arg2) in this case.

File: src/Examples/ExampleFunctionSymbols.cpp [hunk 2]
```diff
@@ -6,6 +6,7 @@
 #include "PDB_RawFile.h"
 #include "PDB_DBIStream.h"
 
+#pragma warning (disable : 4189)
```
[the remark this answers is not in the document]
Comment (MolecularMatters) [reply src/Examples/ExampleFunctionSymbols.cpp:9]: Same here.
Please don't disable warnings like this, but fix the issue instead.

File: src/Examples/ExampleMemoryMappedFile.cpp [hunk 3]
```diff
@@ -31,16 +35,50 @@ MemoryMappedFile::Handle MemoryMappedFile::Open(const wchar_t* path)
...
+	int file = open(path, O_RDONLY);
+
+	if (file == INVALID_HANDLE_VALUE)
+	{
+		return Handle { INVALID_HANDLE_VALUE, nullptr, 0 };
+	}
+
+	if (fstat(file, &fileSb) == -1)
+	{
+		return Handle { INVALID_HANDLE_VALUE, nullptr, 0 };
```
Comment (MolecularMatters) [inline src/Examples/ExampleMemoryMappedFile.cpp:50]: Doesn't this leak the file handle here?

File: src/PDB_ModuleInfoStream.cpp [hunk 4]
```diff
@@ -137,7 +137,7 @@ PDB::ModuleInfoStream::ModuleInfoStream(const DirectMSFStream& directStream, uin
 		streamOffset += objectNameLength + 1u;
 
 		// the stream is aligned to 4 bytes
-		streamOffset = BitUtil::RoundUpToMultiple(streamOffset, 4ull);
+		streamOffset = BitUtil::RoundUpToMultiple(streamOffset, static_cast<size_t>(4ul));
```
Comment (MolecularMatters) [inline src/PDB_ModuleInfoStream.cpp:140]: I prefer using an explicit template argument in that case, i.e. BitUtil::RoundUpToMultiple<size_t>(arg1, arg2).

File: src/PDB_Types.h [hunk 5]
```diff
@@ -14,10 +14,12 @@ namespace PDB
 	// emulating std::byte from C++17 to make the intention clear that we're dealing with untyped data in certain cases, without actually requiring C++17
 	enum class Byte : unsigned char {};
 
+	typedef void* HANDLE;
+
 	// this matches the definition in guiddef.h, but we don't want to pull that in
 	struct GUID
```
Comment (MolecularMatters) [inline src/PDB_Types.h:20]: I think a better solution would be to use the uint*_t types here. This is less guesswork for the reader as to how large a struct is, and should make it more compatible with other platforms & compilers.

**Review by MolecularMatters — CHANGES_REQUESTED** [review]
Thank you so much for providing Linux support.
I'm terribly sorry that the review took me so long, I somehow completely missed this!

File: src/Examples/ExampleFunctionSymbols.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (stevenhoving) [reply src/Examples/ExampleFunctionSymbols.cpp:9]: indeed

File: CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.16)
+
+project(raw_pdb)
+
+set(CMAKE_CXX_STANDARD 17)
```
Comment (MolecularMatters) [inline CMakeLists.txt:5]: The library should only need C++11, not 17.

File: src/PDB_Types.h [hunk 7]
```diff
@@ -159,4 +159,4 @@ namespace PDB
 		uint32_t offset;		// offset into the symbol record stream
 		uint32_t cref;
 	};
-}
```
Comment (MolecularMatters) [inline src/PDB_Types.h:162]: The GUID definition still needs fixing by switching to the uint*_t types.

**Review by MolecularMatters — COMMENTED** [review]
Thanks for fixing most of the issues.
There are only minor things left before this can be merged.

**Review by MolecularMatters — APPROVED** [review]
Looks good, thank you again!

**Pull request merged**